src/backend/chat/tools/fetch_url.py

```diff
@@ -0,0 +1,420 @@
+                        # returning everything inline.
+                        if conversation and user and len(content) > MAX_INLINE_CONTENT_CHARS:
+                            parsed = await _store_in_rag_and_attachments(
+                                conversation=conversation,
+                                user=user,
+                                url=url,
+                                content_bytes=content.encode("utf-8"),
+                                content_type="text/markdown",
+                            )
+                            preview = parsed[:MAX_INLINE_CONTENT_CHARS]
+                            return ToolReturn(
+                                return_value={
+                                    "url": url,
+                                    "original_url": url,
+                                    "stored_in_rag": True,
+                                    "content_preview": preview,
+                                    "source": DOCS_HOST,
+                                    "content":(
+                                    "Le contenu de ce document est volumineux et a été indexé dans "
+                                    "la base de documents de la conversation. "
+                                    "Pour l’interroger, tu dois utiliser l’outil `document_search_rag` "
+                                    "avec une requête précise décrivant ce que tu cherches dans ce document."
+                                )
+                                },
+                                metadata={"sources": {url}},
+                            )
+
+                        return ToolReturn(
+                            return_value={
+                                "url": url,
+                                "original_url": url,
+                                "content": content[:MAX_INLINE_CONTENT_CHARS],
+                                "source": DOCS_HOST,
+                            }
+                        )
+                except Exception as e:
+                    logger.warning("Error fetching Docs content %s: %s", url, e)
+                    return ToolReturn(
+                        return_value={"url": url, "error": str(e)},
+                        content="Ce document Docs n'est pas public ou une erreur est survenue."
+                    )
+
+        async with httpx.AsyncClient(timeout=30.0, follow_redirects=True) as client:
+            response = await _get_with_retry(client, url)
+            extracted = trafilatura.extract(response.text) or response.text
+            content_type_header = response.headers.get("content-type", "unknown")
+            content_type = content_type_header.split(";", 1)[0].strip().lower()
+
+            is_binary_like = not content_type.startswith("text/")
+            is_pdf = "pdf" in content_type or url.lower().endswith(".pdf")
+
+            # For large or binary/PDF content, store in RAG instead of returning everything inline.
```

Comment:
We should always use the RAG, also the "document converter" should be use to detect MIME type I think.